Clockwise.Redis/CircuitBreakerStoragePartition.cs

```diff
@@ -0,0 +1,167 @@
+            await TransitionStateTo(openState, expiry);
+        }
+
+        public async Task SignalSuccessAsync() 
+        {
+            var target = CircuitBreakerState.Closed;
+            if (stateDescriptor?.State == CircuitBreakerState.Open)
+            {
+                target = CircuitBreakerState.HalfOpen;
+            }
+            var targetState = new CircuitBreakerStateDescriptor(target, Clock.Current.Now());
+            await TransitionStateTo(targetState);
+        }
+
+        public async Task<CircuitBreakerStateDescriptor> GetLastStateAsync()
+        {
+            var serialised = await db.StringGetAsync(key);
+            lastSerialisedState = serialised.HasValue ? serialised.ToString() : Empty;
+            if (IsNullOrWhiteSpace(serialised))
+            {
+                var newState = new CircuitBreakerStateDescriptor(CircuitBreakerState.Closed, Clock.Current.Now(),
+                    TimeSpan.FromMinutes(1));
+                lastSerialisedState = await Transistion(lastSerialisedState,
+                    JsonConvert.SerializeObject(newState, JsonSerializationSettings));
+            }
+
+            return TryDeserialise(lastSerialisedState);
+        }
+
+        private static CircuitBreakerStateDescriptor TryDeserialise(string serialised)
+        {
+            return IsNullOrWhiteSpace(serialised)
+                ? null
+                : JsonConvert.DeserializeObject<CircuitBreakerStateDescriptor>(serialised, JsonSerializationSettings);
+        }
+        private async Task TransitionStateTo(CircuitBreakerStateDescriptor targetState, TimeSpan? expiry = null)
+        {
+            var serialsied = JsonConvert.SerializeObject(targetState, JsonSerializationSettings);
+            var stateExpiry = targetState.State == CircuitBreakerState.Open && expiry == null
+                ? TimeSpan.FromMinutes(1)
+                : expiry;
+            await Transistion(lastSerialisedState, serialsied, stateExpiry);
+        }
+        private async Task<string> Transistion(string fromState, string toState, TimeSpan? newStateExpiry = null)
+        {
+            var setCommand = newStateExpiry == null ? $"redis.call(\'set\',\'{key}\',\'{toState}\')" : $"redis.call(\'setex\',\'{key}\', {newStateExpiry.Value.TotalSeconds},\'{toState}\' )";
+            var script = $"local prev = redis.call(\'get\', \'{key}\') if not(prev) or prev == \'{fromState}\' then {setCommand} return \'{toState}\' else return prev end";
+            var execution = (await db.ScriptEvaluateAsync(script))?.ToString();
+            return execution;
+        }
+        private async Task<CircuitBreakerStateDescriptor> ReadDescriptor()
+        {
+            var desc = new CircuitBreakerStateDescriptor(CircuitBreakerState.Closed, Clock.Now(),
+                TimeSpan.FromMinutes(1));
+            var src = await db.StringGetAsync(key);
+
+            if (!src.IsNullOrEmpty)
```

Comment:
`IsNullOrWhitespace` would be safer.

Reply:
that is a redis property, not the static method of string